pkg/utils/validator/resourceplacement.go

```diff
@@ -509,3 +549,64 @@ func supportedResourceCapacityTypes() []string {
 	sort.Strings(capacityTypes)
 	return capacityTypes
 }
+
+type ResourcePlacementObject interface {
```

Comment:
what is this for? We already have placementObj as the unified interface for CPR and RP? Why create another?